web-ng/src/app/shared/models/project.model.ts

```diff
@@ -14,11 +14,32 @@
  * limitations under the License.
  */
 
-import { StringDictionary } from "./string-dictionary.model";
+import { Layer } from './layer.model';
+import { List } from 'immutable';
+import { StringMap } from './string-map.model';
 
 export class Project {
   constructor(
-    readonly title: StringDictionary,
-    readonly description: StringDictionary
+    readonly id: string,
+    readonly title: StringMap,
+    readonly description: StringMap,
+    readonly layers: List<Layer>
   ) {}
+
+  /**
+   * Converts the raw object representation deserialized from JSON into an
+   * immutable Project instance.
+   *
+   * @param id the uuid of the project instance.
+   * @param data the source data in a dictionary keyed by string.
+   */
+
+  static fromJson(id: string, data: { [key: string]: any }): Project {
+    return new Project(
+      id,
+      StringMap(data.title),
```

Comment:
I am not very familiar with the language, but if we put just `Map<string, string>` instead of `StringMap` here (and everywhere else presumably), will it still work? If it still works, I think having `StringMap` as additional layer wouldn't help much here. If not, it looks fine to me.